src/shared/script/shared-code/suffixUtils.ts

```diff
@@ -0,0 +1,305 @@
+import { PLNumber, PLFractionNumber, plString, PLString } from 'pocket-lisp-stdlib'
+import { assertInteger, typeCheck } from './utils'
+import { baseNum2text } from './langUtils'
+
+export function getLastDigit(num: number): number {
+  const n = Math.abs(num)
+  return n % 10
+}
+
+export function getLastNonZeroDigit(num: number): number {
+  if (num === 0) {
+    return 0
+  } else {
+    let lastDigit = getLastDigit(num)
+    while (lastDigit === 0) {
+      num = num / 10
+      lastDigit = getLastDigit(num)
+    }
+    return lastDigit
+  }
+}
+
+export function trailingZeros(num: number): number {
+  // number of trailing zeros after first digit
+  assertInteger(num)
+  if (Math.abs(num) < 10) {
+    return 0
+  }
+  let nZeros = 0
+  while (num % 10 === 0) {
+    num /= 10
+    nZeros++
+  }
+  return nZeros
+}
+
+export function isNumberGroupEven(n: number): boolean {
+  // big numbers (see: https://hu.wikipedia.org/wiki/T%C3%ADz_hatv%C3%A1nyai)
+  const nZeros = trailingZeros(n)
+  const nGroups = Math.floor(nZeros / 3)
+  return nGroups % 2 === 0
+}
+
+export function pitch(n: number): string {
+  // check if number is high pitch (e/i) or low pitch (a/o/u)
+  assertInteger(n)
+  const nZeros = trailingZeros(n)
+  if (nZeros <= 1) {
+    const highPitch = nZeros === 0 ? [1, 2, 4, 5, 7, 9] : [1, 4, 5, 7, 9]
+    const lastDigit = getLastNonZeroDigit(n)
+    return highPitch.indexOf(lastDigit) >= 0 ? 'high' : 'low'
```

Comment:
better to use `includes` instead of `indexOf` to check if the item is in the array

```suggestion
    return highPitch.includes(lastDigit) ? 'high' : 'low'
```

Reply:
done.